### ISSUE TITLE
 Attempt to invoke virtual method 'java.lang.String org.altbeacon.beacon.Identifier.toString()' on a null object reference

### ISSUE THREAD

**Issue by affibox** [mmazzarolo/react-native-beacons-android#23]
``` javascript
09-16 10:27:56.393 16600-16714/com.ruralking2 E/AndroidRuntime: FATAL EXCEPTION: IntentService[BeaconIntentProcessor]
                                                                Process: com.ruralking2, PID: 16600
                                                                java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.altbeacon.beacon.Identifier.toString()' on a null object reference
                                                                    at com.mmazzarolo.beaconsandroid.BeaconsAndroidModule.createRangingResponse(BeaconsAndroidModule.java:233)
                                                                    at com.mmazzarolo.beaconsandroid.BeaconsAndroidModule.access$300(BeaconsAndroidModule.java:35)
                                                                    at com.mmazzarolo.beaconsandroid.BeaconsAndroidModule$2.didRangeBeaconsInRegion(BeaconsAndroidModule.java:226)
                                                                    at org.altbeacon.beacon.BeaconIntentProcessor.onHandleIntent(BeaconIntentProcessor.java:65)
                                                                    at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
                                                                    at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                    at android.os.Looper.loop(Looper.java:145)
                                                                    at android.os.HandlerThread.run(HandlerThread.java:61)
```

**Comment by mmazzarolo** [comment]
Hi! Can you detail it a bit more? What where you trying to do when you received this error?

**Comment by affibox** [comment]
FYI: all i did was follow your example here: https://www.npmjs.com/package/react-native-beacons-android

**Comment by affibox** [comment]
I was doing what you say we should do on the npmjs page

**Comment by mmazzarolo** [comment]
What version o react-native and react-native-beacons-android are you using?

**Comment by affibox** [comment]
Also. Its a new project so no other dependencies where added but yours.

**Comment by affibox** [comment]
"dependencies": {
    "react": "15.3.1",
    "react-native": "0.33.0",
    "react-native-beacons-android": "^2.0.0"
  }

**Comment by mmazzarolo** [comment]
You're right, my mistake.  
If you want to fix it right now you can replace [line 233 of `android/src/main/java/com/mmazzarolo/beaconsandroid/BeaconsAndroidModule.java`](https://github.com/mmazzarolo/react-native-beacons-android/blob/master/android/src/main/java/com/mmazzarolo/beaconsandroid/BeaconsAndroidModule.java#L233) with:  

``` java
map.putString("uuid", region.getId1() != null ? region.getId1().toString() : "");
```

Otherwise you can just provide a beacon uuid to the the `startRangingBeaconsInRegion` function (just take a look at the function API in the README.md).   

If you can give me a feedback I'll try to update the library this evening, otherwise I'll update it on Monday (because unfortunately I don't have any beacon to test it).  

Thanks!

**Comment by affibox** [comment]
It worked. But now this won't work at all and it ended crashing the app:
DeviceEventEmitter.addListener('beaconsDidRange', (data) => {
  console.log('Found beacons!', data.beacons)
});

I can see the beacon I have on android studio but no on the console log;
From Android Studio: 09-16 11:09:39.813 20081-26512/com.ruralking2 D/BeaconsAndroidModule: rangingConsumer didRangeBeaconsInRegion, beacons: [id1: f06da7a8-8652-11e5-af63-feff819cdc9f id2: 777 id3: 22224]

Console log:
Console was cleared
C:\affibox\RuralKing2\node_modules\react-native\Libraries\Utilities\infoLog.js:17 Running application "RuralKing2" with appParams: {"initialProps":{},"rootTag":1}. **DEV** === true, development-level warning are ON, performance optimizations are OFF
C:\affibox\RuralKing2\index.android.js:22 Beacons ranging started succesfully!

**Comment by affibox** [comment]
09-16 11:09:42.533 20081-26646/com.ruralking2 E/art: Throwing OutOfMemoryError "pthread_create (1040KB stack) failed: Try again"
09-16 11:09:42.543 20081-26646/com.ruralking2 E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
                                                                Process: com.ruralking2, PID: 20081
                                                                java.lang.OutOfMemoryError: pthread_create (1040KB stack) failed: Try again
                                                                    at java.lang.Thread.nativeCreate(Native Method)
                                                                    at java.lang.Thread.start(Thread.java:1063)
                                                                    at java.util.concurrent.ThreadPoolExecutor.addWorker(ThreadPoolExecutor.java:920)
                                                                    at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1338)
                                                                    at okhttp3.ConnectionPool.put(ConnectionPool.java:135)
                                                                    at okhttp3.OkHttpClient$1.put(OkHttpClient.java:149)
                                                                    at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:188)
                                                                    at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:129)
                                                                    at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:98)
                                                                    at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
                                                                    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
                                                                    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
                                                                    at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:109)
                                                                    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
                                                                    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
                                                                    at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
                                                                    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
                                                                    at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:124)
                                                                    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
                                                                    at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
                                                                    at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:170)
                                                                    at okhttp3.RealCall.access$100(RealCall.java:33)
                                                                    at okhttp3.RealCall$AsyncCall.execute(RealCall.java:120)
                                                                    at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
                                                                    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                                                                    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                                                                    at java.lang.Thread.run(Thread.java:818)

**Comment by affibox** [comment]
After a few seconds the app crashes.

**Comment by affibox** [comment]
Did you fixed or know why the app is crashing?

**Comment by mmazzarolo** [comment]
Oh, you edited the previous post, you said it was working, right?
I'll take a look into it.  
You're using it on a real device, not on emulator, right?

**Comment by affibox** [comment]
Yes. Real device.  Samsung Note 3

**Comment by mmazzarolo** [comment]
In the meanwhile you can provide an uuid to the function as the second parameter, which is how the ranging should be used most of the time :)

**Comment by affibox** [comment]
did you found anything regarding this issue?

**Comment by mmazzarolo** [comment]
I'm sorry man but I'm really busy with my job and other stuff... 
I'd love having a bit more time to put in this project but it is really time consuming because you can't just sit down and code, you need to test it on real devices and with beacons.  
Please be patient.  

Also, you already have two available solution to this issue: 
1. Did you already try providing the beacon UUID to the `startRangingBeaconsInRegion`? This is the way the function should be used and it works (everyone else use this, way, give it a try!)
2. If you really need to scan for every single close beacons... you can downgrade `react-native@1.3.1` and try again, it should work!  

Let me know if any of this solutions work for you :)

**Comment by affibox** [comment]
Sorry if i upset you. FYI, the app is working and not crashing for now. We are still doing some tests to make sure it stays working. 

I updated Android studio and with it a bunch a files on the project. Maybe that helped with the crashing.

**Comment by mmazzarolo** [comment]
The fix I proposed above is not available in `react-native-beacons-android@2.1.0`, let me know if you have any other issues :)

**Comment by affibox** [comment]
The error for now, was an issue with the J's debugger. Where you try to log the beacon detection on the js debugger on chrome, it crashes the app.
